src/wallet/scriptpubkeyman.cpp

```diff
@@ -2180,6 +2180,19 @@ TransactionError DescriptorScriptPubKeyMan::FillPSBT(PartiallySignedTransaction&
                     *keys = Merge(*keys, *pk_keys);
                 }
             }
+            for (const auto& pk_pair : input.m_tap_bip32_paths) {
+                const XOnlyPubKey& pubkey = pk_pair.first;
+                for (unsigned char prefix : {0x02, 0x03}) {
+                    unsigned char b[33] = {prefix};
+                    std::copy(pubkey.begin(), pubkey.end(), b + 1);
+                    CPubKey fullpubkey;
+                    fullpubkey.Set(b, b + 33);
+                    std::unique_ptr<FlatSigningProvider> pk_keys = GetSigningProvider(fullpubkey);
```

Comment:
58a1a0c7a3f813c887d349ea3670584aff3773ad : we could probably use a `GetSigningProvider` specifically for `XOnlyPubKey`, but that can wait.